Added content for redis installation and configuration (#341)
developer-portal/content: tech/database/redis/about.md

---
title: Redis
subsection: redis
section: tech-database
description: In-memory data structure store, used as a database, cache and message broker
---

# Redis

Redis is a free and open source in memory data structure store. It is widely used as a database, in memory cache or a message broker.

> # Quick start

Reviewer: Please use the Second level header  (`## Quick start`), the title should be only one.
```suggestion
## Quick start
```

> Redis installation on fedora

Reviewer: Let's be a bit more verbose:
```suggestion
To install Redis on Fedora, run the following commands:
```

```sh
$ sudo dnf install redis     # Install redis cli and server
$ sudo systemctl start redis # Initialize redis server
```

To start redis on boot, run

```sh
sudo systemctl enable redis
```

> To test, run

Reviewer: I am not sure what this is testing, my installation, am I testing the CLI client?

```sh

> $ redis-cli ping # output will be pong

Reviewer: Consider adding the expected output under this command and with a description along the lines of `Result of running the test looks like this`.

```

# Configuring Redis

The config file for redis is located at `/etc/redis.conf`.

```sh

> sudo nano /etc/redis.conf

Reviewer: Fix the configuration file path to be `/etc/redis/redis.conf`.

```

> Redis should be always restarted after changing settings. Restart redis by running

Reviewer: From Contributing guidelines: [Be sure](https://developer-portal.github.io/contributing/write-general-writing-hints.html#be-sure)

```suggestion
Redis must always be restarted after changing settings. Restart Redis by running:
```

```sh
sudo systemctl restart redis
```

> Secure redis by enabling authentication. Add this line in redis.conf

Reviewer: It would be better if the `redis.conf` was a full path, i.e. `/etc/redis/redis.conf`

```sh
requirepass  <AuthPassword>
```

> and after redis server restart, all clients must execute `AUTH <AuthPassword>` to execute commands in redis.

Reviewer: ```suggestion
and after the Redis server restarts, all clients must execute `AUTH <AuthPassword>` to be able to execute commands in Redis.
```
